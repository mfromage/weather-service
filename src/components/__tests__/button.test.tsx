import Button from '@/components/button';
import {fireEvent, render} from '@testing-library/react-native';

describe('Button', () => {
  const setup = (props = {}) => {
    const testId = 'button-test-id';
    const defaultProps = {title: 'Hello world', onPress: jest.fn()};
    const utils = render(
      <Button {...defaultProps} {...props} testID={testId} />,
    );
    const button = utils.getByTestId(testId);
    return {...utils, button};
  };
  it('render with given props', () => {
    const title = 'button 1';
    const {getByText} = setup({title});
    expect(getByText(title)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const {toJSON} = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('call onPress', () => {
    const onPress = jest.fn();
    const {button} = setup({onPress});

    fireEvent.press(button);
    expect(onPress).toHaveBeenCalled();
  });
});
