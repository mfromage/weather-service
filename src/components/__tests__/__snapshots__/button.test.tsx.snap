// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 12,
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
  testID="button-test-id"
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    Hello world
  </Text>
</View>
`;
